test/tests/materials/coupled_value_function/adjac.i

```diff
@@ -0,0 +1,91 @@
+[Mesh]
+  [gen]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 5
+    ny = 5
+  []
+[]
+
+[Variables]
+  [u]
+    initial_condition = 0.1
+  []
+  [v]
+    initial_condition = 0.1
+  []
+[]
+
+[Materials]
+  [Du]
+    type = ADCoupledValueFunctionMaterial
+    function = x
+    v = v
+    prop_name = Du
+  []
```

Comment:
I don't think we can pull a material like this from a mortar constraint object.

Reply:
Nothing. Just mentioning that per our conversation on using functions directly, instead of materials, to include derivative information with multivariable interpolation.

Reply:
Oh, yeah, you can just couple a function into a constraint and evaluate it with coupled variables.